gomezmoreno-tp-biblioteca/app.js

```diff
@@ -0,0 +1,253 @@
+        if(Number.isInteger(cantidadNueva) == true && cantidadNueva > 0) {
+            if(cantidadNueva > libros[libroAModificar].disponibles(prestamos)) {
+                libros[libroAModificar].cantidad = cantidadNueva;
+                res.status(204).json("idLibro: " + libroAModificar);
+            }
+            else {
+                res.status(400).json("No puede ingresarse una cantidad menor a la cantidad de ejemplares prestados");
+            }
+        }
+        else {
+            res.status(400).json("La cantidad debe ser un entero mayor a cero");
+        }
+    }
+    else {
+        res.status(404).json("El id del libro ingresado no existe");
+    }  
+})
+
+//RQ1.4: Obtener la cantidad de ejemplares disponibles para prestamo en la libreria
+app.get('/libros/:libro', function (req, res) {
+    var libroAConsultar = encontrarIndice(libros, req.params.libro);
+    if(libroAConsultar != -1) {
+        res.status(200);
+        var copiaLibro = Object.assign({}, libros[libroAConsultar]);
+        copiaLibro["disponibles"] = copiaLibro.disponibles(prestamos);
+        res.json(copiaLibro);
+    }
+    else {
+        res.status(404).json("El id del libro ingresado no existe");
+    } 
+})
+
+//RQ2.1: Crear socio
+app.post('/socios', function (req, res) {
+    var idSocio = getNewId();
+    var nombreSocio = req.body.nombre;
+    if(nombreSocio != undefined) {
+        socios.push(new Socio(idSocio, nombreSocio));
+        res.status(201).json(idSocio);
+    }
+    else {
+        res.status(400).json("No puede agregarse un socio sin ingresar un nombre");
+    }
+})
+
+function librosPrestados(idSocioPrestamos) {
+    var prestamosSocio = new Array();
+    for (var librosPrestadosASocio of prestamos) {
+        if(librosPrestadosASocio.idSocio == idSocioPrestamos) {
+            var copiaPrestamos = Object.assign({}, librosPrestadosASocio);
+            delete copiaPrestamos.idSocio;
+            prestamosSocio.push(copiaPrestamos);
+        }
+    }
+    if(prestamosSocio.length < 1) {
+        return null;
+    } else {
+        return prestamosSocio;
+    }
+}
+
+//RQ2.2: Obtener los libros prestados al socio con sus fechas de vencimiento
+app.get('/socios/:socio/prestamos', function (req, res) {
+    var socioAConsultar = encontrarIndice(socios, req.params.socio);
+    if(socioAConsultar != -1) {
+        var librosSocio = librosPrestados(req.params.socio);
+
+        if (librosSocio != null) {
+            res.status(200);
+            res.json(librosSocio);
+        }
+        else {
+            res.status(400).json("El socio no tiene libros prestados");
+        }
+    }
+    else {
+        res.status(404).json("El id del socio ingresado no existe");
+    }
+})
+
+function comprobarLibrosVencidos(prestamosSocio){
+    var f = new Date();
+    fechaActual = f.getTime();
+
+    if(prestamosSocio != null){
+        for(var p of prestamosSocio){
+            if(fechaActual > p.fechaVencimiento){
+                return false;
```

Comment:
This will return false on the first ocurrence, this should not be a return

Reply:
Changed this function